### PR TITLE
tests on admin 

### PR DESCRIPTION
**Pull request by Techylite**
#delete tests pending

### CODE REVIEW

**Comment by svenfuchs** [comment]
Wheeeee ... such pull request. Go team @Techylite :heart:

**Comment by zaziemo** [comment]
@Ressy77 @essymo wow, that are many changes ;) Thanks for your work! 
I will try to review it asap and hope to get some help from @tyranja. I will start to make some inline comments.

File: .rspec [hunk 1]
```diff
@@ -0,0 +1,6 @@
+--color
+<<<<<<< HEAD
```
Comment (zaziemo) [inline .rspec]: This seems to be a merge conflict that isn't solved yet. Please have a look on that.

File: Gemfile [hunk 2]
```diff
@@ -35,6 +35,16 @@ gem 'rails_12factor', group: :production
 group :development do
   gem 'letter_opener'
   gem 'quiet_assets' # mutes asset pipeline log messages
+  gem 'better_errors'
```
Comment (zaziemo) [inline Gemfile]: Why did you add all those gems? Which problems are they solving?

File: Gemfile [hunk 3]
```diff
@@ -43,10 +53,16 @@ group :development, :test do
...
+  gem 'launchy'
+
+  gem 'minitest'
+  gem 'test-unit'
+
 end
 
 group :test do
-  gem 'factory_girl_rails', '4.1.0'
+  gem 'factory_girl_rails', '4.1.0', :require => false
```
Comment (zaziemo) [inline Gemfile]: Why did you add `:require => false`?

File: app/controllers/profiles_controller.rb [hunk 4]
```diff
@@ -74,6 +74,7 @@ def require_permission
     # Use callbacks to share common setup or constraints between actions.
     def set_profile
       @profile = Profile.find(params[:id])
+
```
Comment (zaziemo) [inline app/controllers/profiles_controller.rb]: unnecessary empty line, please remove

File: app/models/profile.rb [hunk 5]
```diff
@@ -128,5 +128,7 @@ def as_json(options = {})
   def self.search(query)
     where('firstname ILIKE :query OR lastname ILIKE :query OR twitter ILIKE :query', query: "%#{query}%")
   end
+
```
Comment (zaziemo) [inline app/models/profile.rb]: unnecessary empty lines, please remove

File: app/views/admin/categories/edit.html.erb [hunk 6]
```diff
@@ -9,7 +9,7 @@
 
       <%= f.input :name, label: t(:new_name, scope: "admin.dashboard.categories"), input_html: { class: "mv" }, required: false %>
 
-      <%= f.submit t(:save, scope: "general"), class:"btn btn--small" %>
+      <%= submit_tag t(:save, scope: "general"), class:"btn btn--small" %>
```
Comment (zaziemo) [inline app/views/admin/categories/edit.html.erb]: Why did you change that? Btw. it would be great to have longer commit messages that explain what you had in mind when doing your changes. So everyone can understand what problem you solved and why you decided to do the way you did. This is not easy and we also try hard and get reminded everytime by our coach ;) But one example is this one:

"add map to header
and add toggle map function
to show and load map only when user is clicking on address

(#5813)"

File: app/views/admin/medialinks/edit.html.erb [hunk 7]
```diff
@@ -16,7 +16,7 @@
       <%= f.input :url, as: :string, input_html: { class: 'mb input--width' }  %>
       <%= f.input :title, as: :string, input_html: { class: 'mb input--width' }  %>
       <%= f.input :description, as: :text, input_html: { rows: 10, class: 'mb input--width' }  %>
-      <%= f.submit class: 'btn btn--small' %>
+      <%= Click_button class: 'btn btn--small' %>
```
Comment (zaziemo) [inline app/views/admin/medialinks/edit.html.erb]: same question as above: why did you change this?

File: app/views/devise/sessions/new.html.erb [hunk 8]
```diff
@@ -1,6 +1,10 @@
 <% @target_url = session_path(resource_name)
    @signup = false
    %>
+
```
Comment (zaziemo) [inline app/views/devise/sessions/new.html.erb]: Is this meant to stay or was that only for test reasons?

**Comment by Techylite** [comment]
@zaziemo thank you for this. We will look at the comments, correct and improve on what needs to be improved

File: spec/controllers/edit_profiles_spec.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+require 'spec_helper'
+
+describe "edit_profiles" do
+	context "signed in as admin" do
```
Comment (tyranja) [inline spec/controllers/edit_profiles_spec.rb]: Indentation seems to be wrong.

File: spec/models/profile_spec.rb [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  it { should be_valid }
+
+context 'when firstname is not present' do
+    before { @profile.firstname = '' }
+    it { should_not be_valid }
+end
+
+
+context 'when lastname is not present' do
+	 before { @profile.firstname = " " }
```
Comment (tyranja) [inline spec/models/profile_spec.rb]: Isn't it @profile.lastname? Why different quotes? Indentation seem to be wrong.

File: spec/models/profile_spec.rb [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+ 		 end
+	end
+end
+
+describe "admin? is true when an admin user" do
+	admin = FactoryGirl.create(:admin)
+    profile.admin = true
+    it {should_be_valid}
+   
+  end
```
Comment (tyranja) [inline spec/models/profile_spec.rb]: What are these test for? Maybe you can write better test descriptions? And delete the empty lines please!

**Comment by tyranja** [comment]
Wow such a lot of tests! Great work! :+1:

File: app/views/admin/categories/edit.html.erb (hunk 6 above)
Comment (Techylite) [inline app/views/admin/categories/edit.html.erb]: thank you 
we will do

**Comment by Ressy77** [comment]
@zaziemo we went through our code and we decided to change the little errors before they piled up. 
Just to avoid having recurring errors when we do a new push request

File: Gemfile (hunk 2 above)
Comment (Ressy77) [inline Gemfile]: Gem 'better errors' was for our own local machines to help with the errors. 
Gem 'letter opener' is to test the emails sent by Speakerinnen to anyone who registers
We added gem guard to help with automatic testing

File: app/views/devise/sessions/new.html.erb (hunk 8 above)
Comment (Ressy77) [inline app/views/devise/sessions/new.html.erb]: It was only for test reasons. We changed it

File: Gemfile (hunk 3 above)
Comment (Ressy77) [inline Gemfile]: Changed.

File: spec/models/profile_spec.rb [hunk 12]
```diff
@@ -0,0 +1,45 @@
+ require 'spec_helper'
+
+ describe Profile do
```
Comment (zaziemo) [inline spec/models/profile_spec.rb]: Hey @essymo, I'm currently trying to find out how to make these tests passing. One thing is that an email address can only be used once. So if you create a profile out of the factory you would create two contexts and move the let block separately into each context (so it is only valid in this specific context and the email address can be used in the next context again without problems). See how I refactored the code - now the tests pass: https://gist.github.com/zaziemo/8df5b517a04fac0fe1cd

File: spec/support/utilities_spec.rb [hunk 13]
```diff
@@ -1,3 +1,16 @@
-require 'support/sign_in_helper_spec'
```
Comment (zaziemo) [inline spec/support/utilities_spec.rb]: I'm not sure what these support files do but apparently something goes wrong here and the "sign in" examples in navigation_spec.rb fail. What I found out is, that support/sign_in_helper_spec.rb and support/utilities_spec.rb have the same content. Probably that is what is responsable for the failing tests. Probably you want to check that and discuss it with your coach.

Comment (Techylite) [inline spec/support/utilities_spec.rb]: Will do . That seems to help with a problem we hard.

Comment (essymo) [inline spec/support/utilities_spec.rb]: one problem though...We still get the Capybara::ElementNotFound:
error...tried to include the sign_in_helper but capybara still expects a
button.

On 1 September 2015 at 18:43, techylite <notifications@github.com> wrote:

> In spec/support/utilities_spec.rb
> <https://github.com/rubymonsters/speakerinnen_liste/pull/392#discussion_r38434497>
> :
>
> > @@ -1,3 +1,16 @@
> > -require 'support/sign_in_helper_spec'
>
> Will do . That seems to help with a problem we hard.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rubymonsters/speakerinnen_liste/pull/392/files#r38434497>
> .
>



-- 
Esther Monchari
Business Information Technology
Skype Esther.Monchari

Comment (zaziemo) [inline spec/support/utilities_spec.rb]: Can you find out what Capybara expects from a button element? Eg a certain class of html element? What makes the function click_button working? Probably this Element does not appear in the test environment (then try to make it appear). Or: is this button or the I18n-key still named aus the test suggests?

**Comment by zaziemo** [comment]
Wohooo! All the tests are passing! We will merge them as soon as possible! Thanks for you work @Techylite!

**Comment by zaziemo** [comment]
Hum, apparently they are not passing on my local machine. There is a very strange behaviour. Sometimes I get 8 failures, sometimes 18, sometimes 34. Really strange. Did you have any problems with this before @Techylite?

**Comment by essymo** [comment]
Just for the navigation spec..let me run it again and see

On Friday, 4 September 2015, Maren Heltsche <notifications@github.com>
wrote:

> Hum, apparently they are not passing on my local machine. There is a very
> strange behaviour. Sometimes I get 8 failures, sometimes 18, sometimes 34.
> Really strange. Did you have any problems with this before @Techylite
> <https://github.com/Techylite>?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rubymonsters/speakerinnen_liste/pull/392#issuecomment-137677194>
> .
>


-- 
Esther Monchari
Business Information Technology
Skype Esther.Monchari

**Comment by zaziemo** [comment]
@essymo I found the problem: apparently you deleted some lines in the devise.en.yml which caused the problems that some buttons were not found.

**Comment by essymo** [comment]
or maybe the file(devise.en.yml) was updated,I think.let me try and compare

On 4 September 2015 at 12:39, Maren Heltsche <notifications@github.com>
wrote:

> @essymo <https://github.com/essymo> I found the problem: apparently you
> deleted some lines in the devise.en.yml which caused the problems that some
> buttons were not found.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rubymonsters/speakerinnen_liste/pull/392#issuecomment-137691198>
> .
>



-- 
Esther Monchari
Business Information Technology
Skype Esther.Monchari

**Comment by zaziemo** [comment]
@Techylite ouff - everything merged now and refactored so it works now. Thanks for your contribution!

**Comment by tyranja** [comment]
@Techylite Well done!! :ok_hand:

**Comment by Ressy77** [comment]
Yeeiy :-)
Thanks @zaziemo and @tyranja.

**Comment by essymo** [comment]
Awesome🙌

On 10 September 2015 at 16:51, Resla Wesonga <notifications@github.com>
wrote:

> Yeeiy :-)
> Thanks @zaziemo <https://github.com/zaziemo> and @tyranja
> <https://github.com/tyranja>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rubymonsters/speakerinnen_liste/pull/392#issuecomment-139240613>
> .
>



-- 
Esther Monchari
Business Information Technology
Skype Esther.Monchari
